app/models/herbarium.rb

```diff
@@ -168,19 +168,33 @@ def merge_associated_records(src)
   def self.primer
     result = ""
     if User.current
-      result = connection.select_values(%(
-        SELECT DISTINCT h.name AS x
-        FROM herbarium_records s, herbaria h, herbaria_curators c
-        WHERE s.herbarium_id = h.id
-        AND h.id = c.herbarium_id
-        AND c.user_id = #{user_id}
-        ORDER BY s.updated_at DESC
-        LIMIT 100
-      )).sort
+      select_manager = arel_select_herbarium_primer
+      # puts(select_manager.to_sql)
+      result = connection.select_values(select_manager).sort
     end
     result
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # SELECT DISTINCT h.name AS x
+  # FROM herbarium_records s, herbaria h, herbaria_curators c
+  # WHERE s.herbarium_id = h.id
+  # AND h.id = c.herbarium_id
+  # AND c.user_id = #{user_id}
+  # ORDER BY s.updated_at DESC
+  # LIMIT 100
+  private_class_method def self.arel_select_herbarium_primer
+    s = HerbariumRecord.arel_table
+    h = Herbarium.arel_table
+    c = Arel::Table.new(:herbaria_curators)
+    # user_id = 252 # for testing
+
+    h.join(s).on(s[:herbarium_id].eq(h[:id])).join(c).on(
+      h[:id].eq(c[:herbarium_id]).and(c[:user_id].eq(user_id))
+    ).project(h[:name]).distinct.order(s[:updated_at]).take(100)
+  end
+  # rubocop:enable Metrics/AbcSize
```

Comment:
How about the following? 
It's just ActiveRecord. (I probably should have figure this out a long time ago.) Is there an advantage to using AREL if ActiveRecord does the trick?
@pellaea: Is it correct to return a an empty string (as opposed to an empty list) if there's no current user?
```suggestion
  def self.primer
    return "" unless (user_id = User.current_id)

    Herbarium.joins(:herbaria_curators, :herbarium_records).
      where(herbaria_curators: {user_id: user_id}).distinct.
      order(updated_at: :desc).limit(100).pluck(:name)
end
```